src/libcore/char/methods.rs

```diff
@@ -1245,7 +1245,14 @@ impl char {
     #[stable(feature = "ascii_ctype_on_intrinsics", since = "1.24.0")]
     #[inline]
     pub fn is_ascii_hexdigit(&self) -> bool {
-        self.is_ascii() && (*self as u8).is_ascii_hexdigit()
+        if !self.is_ascii() {
```

Comment:
Why not change this like is_ascii_digit to `self.is_digit(16)`? Does that end up with worse codegen?

Reply:
yes, currently is_digit(x) checks two separate alphabetic ranges instead of one bit-manipulated one, but i could change that if desired

Reply:
Yeah, seems like it'd be best to fix is_digit.